x-pack/plugins/ml/public/services/results_service.js

```diff
@@ -1391,6 +1392,147 @@ function getEventRateData(
   });
 }
 
+// Queries Elasticsearch to obtain event distribution i.e. the count
+// of entities over time.
+// index can be a String, or String[], of index names to search.
+// Extra query object can be supplied, or pass null if no additional query.
+// Returned response contains a results property, which is an object
+// of document counts against time (epoch millis).
+// const SAMPLER_TOP_TERMS_THRESHOLD = 100000;
+const SAMPLER_TOP_TERMS_SHARD_SIZE = 200;
+function getEventDistributionData(
```

Comment:
The long-term aim is to switch to using the server-side results service in `ml/server/models/results_service`. Could this be moved to the server-side model (or add to future work PR)?

Reply:
Yes, I think at some point it would actually be good to move the most of what's going on in `explorer_charts_container_service.js` to a Kibana API endpoint, at the moment the charts are doing a lot of individual requests from the client side. Added a note about it in #21163